bin/g5-calc-noise.sh

```diff
@@ -28,73 +27,53 @@ function ReportNoiseAndExit()
 
 if [ -e $INPUT ]; then
   yarr=( $(tail -$MAXRECORDS $INPUT | cut -d ',' -f2 ) )
-  xdate=( $(tail -$MAXRECORDS $INPUT | cut -d ',' -f1 ) )
   n=${#yarr[@]}
 else
-  noise=0
+  noise=0.5
   ReportNoiseAndExit
 fi
 
 #    set initial x values based on date differences
 
 if [ $(bc <<< "$n < $MINRECORDS") -eq 1 ]; then
   # set noise = 0 - unknown
-  noise=0
+  noise=0.5
 	#echo "noise = 0 no records"
   ReportNoiseAndExit
 fi
 
-firstDate=${xdate[0]}
-for (( i=0; i<$n; i++ ))
-do
-  xarr[$i]=$(bc -l <<< "(${xdate[$i]} - $firstDate) / 15") 
-#  echo "x,y=${xarr[$i]},${yarr[$i]}"
-done
-
-#echo ${xarr[@]}
-#echo ${xdate[@]}
 #echo ${yarr[@]}
 
 # sod = sum of distances
 sod=0
 overallDistance=0
 
+# add 0.1 for each bounce
+# allow 5 point rises without adding noise
 lastDelta=0
+noise=0
 for (( i=1; i<$n; i++ ))
 do
-  # time-based multiplier 
-  # y2y1Delta adds a multiplier that gives 
-  # higher priority to the latest BG's
-  y2y1Delta=$(bc -l  <<< "(${yarr[$i]} - ${yarr[$i-1]}) * (1 +  $i/($n * 4))")
-  x2x1Delta=$(bc -l <<< "${xarr[$i]} - ${xarr[$i-1]}")
-  #echo "x delta=$x2x1Delta, y delta=$y2y1Delta" 
+  y2y1Delta=$(bc -l  <<< "${yarr[$i]} - ${yarr[$i-1]}")
+  if [ $(bc -l <<< "$y2y1Delta < 0") -eq 1 ]; then
+    y2y1Delta=$(bc -l <<< "0 - $y2y1Delta")
+  fi
+  remainder=$(bc -l <<< "$y2y1Delta - 10")
+  if [ $(bc -l <<< "$remainder > 0") -eq 1 ]; then
+    noise=$(bc -l <<< "$noise + $remainder/200") 
+  fi
+  
```

Comment:
no more weighting for the most recent #'s?

Reply:
Yes, I will add that back in to this algorithm.